mdoc-js/src/main/scala/mdoc/modifiers/ScalajsWorker.scala

```diff
@@ -0,0 +1,106 @@
+package mdoc.modifiers
+
+import org.scalajs.linker.interface._
+import java.net.{URL, URLClassLoader}
+import scala.concurrent.Future
+import java.nio.file.{Path, Paths}
+import scala.concurrent.ExecutionContext
+import scala.language.reflectiveCalls
+
+class ScalajsWorkerState(
+    cachedIRFiles: Seq[IRFile],
+    linker: ClearableLinker
+)
+
+class ScalajsWorker(compilerClasspath: Array[URL], claspath: Array[URL]) {
+  private val shared = getClass().getClassLoader()
+
+  private final class FilteringClassLoader(parent: ClassLoader) extends ClassLoader(parent) {
+    private val parentPrefixes = List(
+      "java.",
+      "scala.",
+      "org.scalajs.linker.",
```

Comment:
This should be
```suggestion
      "org.scalajs.linker.interface.",
```
At least, with the architecture where scalajs-linker-interface is still part of the direct dependencies of mdoc, this is what you need to do.

Reply:
I'll give it a go, thanks!

That said, I might as well just remove it and do everything with java reflection directly, without structural types. 
Pretty sure most of the troubles I'm having are because of this

Reply:
Yes. I think, given what you're trying to achieve, that the interface/linker decomposition is not really going to help. I think what you want is to load *everything* related to the linker in a class loader, and interact with it exclusively with Java reflection, indeed.

It will be cumbersome for the `LinkerConfig`, but I don't think there's really any other way. It's not impossible that there would be value in publishing such a version-agnostic bridge as a separate library, written exclusively in Java...

Reply:
Ah dang, the `StandardConfig` is exactly why I wanted to have linker interface in there :-/ That's fine, not the end of the world.

Things would indeed be simpler if there was a Java interface available for this, and as @lolgab showed with some trickery one could use newer version of the interface to invoke older version of Scala.js.. Although outside of build tools I don't know if it's such a valuable thing.